lua/tardis/cl_menuoptions.lua

```diff
@@ -0,0 +1,201 @@
+                        panel:AddItem(el1)
+                        if el2 then panel:AddItem(el2) end
+                    end
+
+                    if data.subsection and subsections[section]
+                        and subsections[section][data.subsection]
+                    then
+                        local unfolded_subsections = TARDIS:GetSetting("options-unfolded-subsections")
+                        local expanded = false
+                        if unfolded_subsections[section] then
+                            expanded = unfolded_subsections[section][data.subsection]
+                        end
+
+                        local subsection = vgui.Create("DForm")
+                        subsection:SetLabel(data.subsection)
+
+                        -- save the subsection state for more convenience
+                        subsection:SetExpanded(expanded or false)
+                        subsection.OnToggle = function(self, expanded)
+                            local unfolded_subsections = TARDIS:GetSetting("options-unfolded-subsections")
+                            unfolded_subsections[section] = unfolded_subsections[section] or {}
+                            unfolded_subsections[section][data.subsection] = expanded
+                            TARDIS:SetSetting("options-unfolded-subsections", unfolded_subsections)
+                        end
+
+                        for a2,b2 in ipairs(options) do
+                            local id2,data2 = b2[1],b2[2]
+                            if data2.section == section and data2.subsection == data.subsection then
+                                local el1,el2 = TARDIS:CreateOptionInterface(id2, data2)
+                                subsection:AddItem(el1)
+                                if el2 then subsection:AddItem(el2) end
+                            end
+                        end
+
+                        panel:AddItem(subsection)
+
+                        local spacer = vgui.Create("DPanel")
+                        spacer:SetTall(2)
+                        panel:AddItem(spacer)
+
+                        subsections[section][data.subsection] = false
+                    end
+                end
+            end
+
+            local reset_button = vgui.Create("DButton")
+            reset_button:SetText("Reset settings (this section)")
+            reset_button.DoClick = function(self)
+                Derma_Query(
+                    "Reset clientside settings of section \"" .. section .. "\"? This cannot be undone.",
+                    "TARDIS Interface",
+                    "OK", function()
+                        TARDIS:ResetSectionSettings(section)
+                        TARDIS:Message(LocalPlayer(), "TARDIS clientside settings of section \"" .. section ..  "\" have been reset. You may need to respawn the TARDIS for all changes to apply.")
+                        RunConsoleCommand("spawnmenu_reload")
```

Comment:
Now resolved, thanks!

Reply:
Done
I've made the section settings refresh without spawnmenu reload
I'm keeping the hacky method for the global refresh though, since it's much more complicated
You can do it yourself if you want, but I don't think it's worth the effort with global settings